File: Algorithm/python_source/charpter12_string/level2/reverse_string.py
# coding=utf-8
from collections import deque
def reverseString(s):
    if s is None or len(s) == 0:
        return s

    left, right = 0, len(s) - 1
    while left < right:
        s[left], s[right] = s[right], s[left]
        left += 1
        right -= 1

    return s

s = list("Hello World")
reverseString(s)
print(''.join(s))  # 输出：dlroW olleH

File: Algorithm/c_or_cpp/charpter12_string/level2/reverse_string.c
#include <stdio.h>

void reverseString(char* s, int n) {
    if (s == NULL || n == 0) {
        return;
    }

    for (int left = 0, right = n - 1; left < right; ++left, --right) {
        char tmp = s[left];
        s[left] = s[right];
        s[right] = tmp;
    }
}

int main() {
    char s[] = "Hello World";
    int n = sizeof(s) - 1; // 计算字符串的长度，不包含最后的'\0'字符
    reverseString(s, n);
    printf("%s\n", s); // 输出：dlroW olleH
    return 0;
}